lib/Conch/Controller/User.pm

```diff
@@ -442,6 +442,100 @@ sub deactivate ($c) {
 	return $c->status(204);
 }
 
+=head2 get_tokens
+
+Get a list of user tokens. By default only returns active tokens; pass C<?with_expired=1>
+to include expired tokens as well.
+
+Response uses the UserTokens json schema.
+
+=cut
+
+sub get_tokens ($c) {
+    my $rs = $c->db_user_session_tokens->order_by('name');
+    $rs = $rs->unexpired if not $c->req->query_params->param('with_expired') // 0;
+
+    return $c->status(200, [ $rs->all ]);
+}
+
+=head2 create_token
+
+Create a new token, creating a JWT from it.  Response uses the NewUserToken json schema.
+
+=cut
+
+sub create_token ($c) {
+    my $input = $c->validate_input('NewUserToken');
+    return if not $input;
+
+    return $c->status(400, { error => 'name "'.$input->{name}.'" is already in use' })
+        if $c->db_user_session_tokens
+            ->search({ user_id => $c->stash('user_id'), name => $input->{name} })->exists;
+
+    # default expiration: 5 years
+    my $expires_abs = time + (($c->config('jwt') || {})->{custom_token_expiry} // 86400*365*5);
+
+    # TODO: ew ew ew, some duplication with Conch::Controller::Login::_create_jwt.
+    my ($new_db_row, $token) = $c->db_user_session_tokens->generate_for_user(
+        $c->stash('user_id'), $expires_abs, $input->{name});
+
+    my $jwt = Mojo::JWT->new(
+        claims => { uid => $c->stash('user_id'), jti => $token },
+        secret => $c->config('secrets')->[0],
+        expires => $expires_abs,
+    )->encode;
+
+    return $c->status(201, {
+        token => $jwt,
+        $new_db_row->TO_JSON->%*,
+    });
```

Comment:
I would also add a `Location` header to point to the endpoint you can GET the new db row data from.

EDIT: WIth a 201 that _shouldn't_ cause a client to re-direct automatically, but it may be that the world conspires against correctness here.